### DISCUSSION TITLE
mContriever pre-built index for Mr.TyDi datasets

### DISCUSSION THREAD

**Discussion by cramraj8** [castorini/pyserini#1907] [Q&A] [answered]
Hi @lintool ,

I see mContriever pre-built index available for MIRACL benchmark, but not for My.TyDi benchmark. However, mDPR is available for both. Any particular reasons why mContriever for TyDi not available ? If you have already built them, would you mind sharing them.

Thanks,
Ramraj

**Comment by lintool** [comment, marked as answer]
Hi @cramraj8 - we just haven't had time to build them, but there's no reason why you couldn't, and I don't think it'd be too difficult. PRs welcome!

**Reply by cramraj8** [reply]
Got it. Thank you.

**Comment by cramraj8** [comment]
@lintool 
I tried to encode BEIR & Mr.TyDi/ MIRACL datasets using Contriever or mContriever models for indexing. But pyserini.encode throws the following error
`__main__.py encoder: error: argument --encoder-class: invalid choice: 'contriever' (choose from 'dpr', 'bpr', 'tct_colbert', 'ance', 'sentence-transformers', 'auto')`
And I don't see the contriver script inside the encode folder as well. Are there any replacements for this or I should use Tevatron ?
